vizro-core/src/vizro/_vizro.py

```diff
@@ -18,6 +18,7 @@ class Vizro:
 
     _user_assets_folder = Path.cwd() / "assets"
     _lib_assets_folder = os.path.join(os.path.dirname(os.path.abspath(__file__)), "static")
+    _cache_config = {"CACHE_TYPE": "NullCache"}
```

Comment:
I think a user should set cache through `data_manager._cache` directly rather than `Vizro._cache_config`. Please could you try out this?
* remove `Vizro._cache_config`
* move `data_manager._cache.init_app(self.dash.server)` to the beginning of the `Vizro.run` method
* in app.py you'd then do something like this:

```
data_manager["iris"] = ...
app = Vizro().build(...)
data_manager._cache =  Cache(config={'CACHE_TYPE': "FileSystemCache", "CACHE_DIR": "cache"})
app.run()
```

Ideally the user should be able to set the cache even after building the app so long as they do it in time for `run`.

Reply:
Tested with this approach - In the pre_build stage, `self._get_original_data` is called already. If the cache is initiated after the pre_build, flask_caching gives error: "AttributeError: 'Cache' object has no attribute 'app'"